app/javascript/src/components/Dashboard/Notes/NotesList.jsx

```diff
@@ -1,35 +1,103 @@
-import React, { useState } from "react";
+import React from "react";
+
+import EmptyNotesListImage from "images/EmptyNotesList";
+
+import EmptyState from "components/Common/EmptyState";
 
 import Note from "./Note";
-import EditNotePane from "./Pane/Edit";
 
-const NotesList = ({ notes, user, fetchNotes }) => {
-  const [showEditNote, setShowEditNote] = useState(false);
-  const [selectedNote, setSelectedNote] = useState({});
+const DUMMY_TAGS = [
+  "Gettings Started",
+  "Onboarding",
+  "User Flow",
+  "UX",
+  "Bugs",
+  "V2",
+];
+const STATUS = ["Drafted", "Created", "Updated"];
+const NAMES = [
+  "Debra",
+  "Allen",
+  "Gerald",
+  "Harris",
+  "Raymond",
+  "Carter",
+  "Jacqueline",
+  "Torres",
+];
 
-  const onEditClickHandler = selectedNote => {
-    setSelectedNote(selectedNote);
-    setShowEditNote(true);
+const NotesList = ({
+  notes,
+  onEditClick,
+  onDeleteClick,
+  setShowNewNotePane,
+}) => {
+  // Only for data creation purpose
+  const chooseRandomTags = () => {
+    const shuffled = DUMMY_TAGS.sort(() => 0.5 - Math.random());
+    const randomNumber = Math.floor(
+      Math.random() * (DUMMY_TAGS.length - 1) + 1
+    );
+
+    return shuffled.slice(0, randomNumber);
   };
 
+  // Only for data creation purpose
+  const addTagsToNote = allNotes =>
+    allNotes.map(note => {
+      const tags = chooseRandomTags();
+      return { ...note, tags };
+    });
+
+  // Only for data creation purpose
+  const addStatusToNote = allNotes =>
+    allNotes.map(note => {
+      const randomNumber = Math.floor(Math.random() * STATUS.length);
+      return { ...note, status: STATUS[randomNumber] };
+    });
+
+  // Only for data creation purpose
+  const addUserToNote = allNotes =>
```

Comment:
- The component is named `NotesList`. So this ought to `list` notes. That's it. But you are performing other unrelated side effects over here like `adding` etc which goes against keeping things DRY and sensible.
- Move these methods into an `utils.js` file within the closest scope which is `Notes` folder.
- Invoke these methods from the parent component where it makes sense.